spec/stack_master/sso_group_id_finder_spec.rb

```diff
@@ -0,0 +1,98 @@
+RSpec.describe StackMaster::SsoGroupIdFinder do
+  let(:region) { 'us-east-1' }
+  let(:identity_store_id) { 'd-12345678' }
+  let(:group_name) { 'AdminGroup' }
+  let(:group_id) { 'abc-123-group-id' }
+
+  let(:aws_client) { instance_double(Aws::IdentityStore::Client) }
+
+  subject(:finder) do
+    # Ruby 3+ keyword args fix: make sure new accepts keyword args
+    allow(Aws::IdentityStore::Client).to receive(:new).with(hash_including(region: region)).and_return(aws_client)
```

Comment:
Can we put the stubbing in a `before` block?

Reply:
Actually, `Aws::IdentityStore::Client.new` doesn't take keyword args. Rather, it takes a `Hash`.

This should work:
```ruby
allow(Aws::IdentityStore::Client).to receive(:new).with({ region: region }).and_return(aws_client)
```

Reply:
I finally read the error message properly, so I reverted a bunch of unncessary change and updated the calls to pass hashes rather than keywords :smile:

Reply:
[f51bee8](https://github.com/envato/stack_master/pull/390/commits/f51bee82b36543a6237f03bfbcb26fadac4f68c1)
[3bf111c](https://github.com/envato/stack_master/pull/390/commits/3bf111c0e2d5ca4dc9f31d135293876f22d9df01)